packages/pigeon/lib/dart_generator.dart

```diff
@@ -49,7 +49,11 @@ final BinaryMessenger$nullTag _binaryMessenger;
       if (func.argType != 'void') {
         argSignature = '${func.argType} arg';
         sendArgument = 'encoded';
-        encodedDeclaration = 'final Object encoded = arg.encode();';
+        if (customEnumNames.contains(func.argType)) {
+          encodedDeclaration = 'final Object encoded = Map<Object?, Object?>{\'value\': arg.index};';
+        } else {
+          encodedDeclaration = 'final Object encoded = arg.encode();';
+        }
```

Comment:
It looks like you are supporting 'naked' enum arguments?  We don't support naked arguments yet.  You should only have to accomodate converting and unconverting of enums inside the `encode` and `decode` methods.

This should be supported:
```dart
enum State {
  OPEN,
  CLOSED
}

class AddCommand {
  int asset;
  State state;
}

@HostApi
abstract class Api {
  void add(AddCommand command);
}
```

We'll come up with a more wholistic solution for naked arguments.

Reply:
Ahh, I see. I misunderstood this then. I'll go and implement what you described!